File: app/src/main/java/com/example/android_recyclerview_list/ImageListAdapter.java
package com.example.android_recyclerview_list;

import android.app.Activity;
import android.content.Intent;
import android.media.Image;
import android.net.Uri;
import android.support.annotation.NonNull;
import android.support.constraint.ConstraintLayout;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.io.Serializable;
import java.util.ArrayList;


public  class ImageListAdapter extends RecyclerView.Adapter<ImageListAdapter.ViewHolder> implements Serializable {

private ArrayList<StoredImage> entryData;


    public ImageListAdapter(ArrayList<StoredImage> entryData) {
        this.entryData = entryData;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.image_item_layout, parent, false);
        return new ViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(ImageListAdapter.ViewHolder holder, int position) {
        final StoredImage data = entryData.get(position);
        holder.itemImageView.setImageURI(data.getUriPic());
        holder.itemTextView.setText(data.getUriString());
        holder.parentView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent fullIntent = new Intent(v.getContext(), ImageDetails.class);
                fullIntent.putExtra("name",data);
                ((Activity)v.getContext()).startActivityForResult(fullIntent, MainActivity.EDIT_IMAGE_REQUEST_CODE);
            }
        });


    }

    @Override
    public int getItemCount() {
        return this.entryData.size();
    }

class ViewHolder extends RecyclerView.ViewHolder{
    ConstraintLayout parentView;
    ImageView itemImageView;
    TextView itemTextView;

    public ViewHolder(@NonNull View itemView) {
        super(itemView);
        itemTextView = itemView.findViewById(R.id.item_layout_item_name);
        itemImageView = itemView.findViewById(R.id.item_layout_item_image);

        parentView = itemView.findViewById(R.id.item_layout_parent_layout);

    }
}

}

File: app/src/main/java/com/example/recyclerviewlist2/adapter/ImageListAdapter.kt
package com.example.recyclerviewlist2.adapter

import android.media.Image
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.recyclerviewlist2.R
import com.example.recyclerviewlist2.model.Imagedata
import kotlinx.android.synthetic.main.image_item_layout.view.*


class ImageListAdapter (val Image_data: MutableList<Imagedata>):
    RecyclerView.Adapter<ImageListAdapter.ViewHolder>(){

    class ViewHolder(view:View) : RecyclerView.ViewHolder(view) {
        val image: ImageView = view.Picture_image
        val description: TextView = view.Picture_Description
        val name: TextView = view.Picture_Name



    }
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int):
            ViewHolder {
        val viewGroup = LayoutInflater.from(parent.context).inflate(R.layout.image_item_layout,parent, false)
        return ViewHolder(viewGroup)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
       holder.description.text = Image_data[position].description
        holder.image.setImageDrawable(Image_data[position].image)
        holder.name.text = Image_data[position].name
    }

    override fun getItemCount(): Int {
        return Image_data.size
    }
}
